tracing-subscriber/src/fmt/mod.rs

```diff
@@ -1131,7 +1134,22 @@ pub fn try_init() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
     #[cfg(feature = "env-filter")]
     let builder = builder.with_env_filter(crate::EnvFilter::from_default_env());
 
-    builder.try_init()
+    let subscriber = builder.finish();
+    #[cfg(not(feature = "env-filter"))]
+        let targets = match std::env::var("RUST_LOG") {
+        Ok(var) => {
+            crate::filter::Targets::from_str(&var).map_err(|e|{eprintln!("Ignoring `RUST_LOG`: {:?}", e);}).unwrap_or_default()
+        }
+        Err(std::env::VarError::NotPresent) => {crate::filter::Targets::new()}
+        Err(e) => {
+            eprintln!("Ignoring `RUST_LOG`: {:?}", e);
+            crate::filter::Targets::new()
+        }
+    }.with_default(crate::fmt::Subscriber::DEFAULT_MAX_LEVEL);
+    #[cfg(not(feature = "env-filter"))]
+    let subscriber = subscriber.with(targets);
```

Comment:
style nit, take it or leave it: it might be slightly neater to place all of the feature-flagged code in a block, so that we only have one `cfg` attribute. this would also allow us to use imports inside that block, so we wouldn't need to use fully qualified paths for everything:
```suggestion
    #[cfg(not(feature = "env-filter"))]
    let subscriber = {
        use std::{env, str::FromStr};
        use crate::{filter::Targets, fmt::Subscriber::DEFAULT_MAX_LEVEL};
        let targets = match env::var("RUST_LOG") {
            Ok(var) => Targets::from_str(&var).map_err(|e|{eprintln!("Ignoring `RUST_LOG={:?}`: {}", var, e);}).unwrap_or_default(),
            Err(env::VarError::NotPresent) => Targets::new().with_default(DEFAULT_MAX_LEVEL),
            Err(e) => {
                eprintln!("Ignoring `RUST_LOG`: {}", e);
                crate::filter::Targets::new().with_default(DEFAULT_MAX_LEVEL)
            },
        };
        subscriber.with(targets)
    };
```